File: Section Reports/Layout/Stylesheets/VB.NET/Stylesheets/StyleSheetsForm.vb
Imports System.IO
Public Class StyleSheetsForm
	Private externalStyleSheet As String = ""
	Private Sub buttonChooseExtStyle_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonChooseExtStyle.Click
		'Select an external stylsheet to apply.
		Dim openFileDialog As FileDialog = New OpenFileDialog()
		
		openFileDialog.Filter = "Style sheet|*.reportstyle"
		
		'
		openFileDialog.InitialDirectory = New FileInfo(Me.GetType().Assembly.Location).DirectoryName
		openFileDialog.CheckFileExists = True
		If openFileDialog.ShowDialog(Me) = DialogResult.OK Then
			Dim styleSheetFile As FileInfo = New FileInfo(openFileDialog.FileName)
			externalStyleSheet = styleSheetFile.FullName
			
			radioButtonExternalStyleSheet.Text = "External stylesheet: " + styleSheetFile.Name
			
			'
			radioButtonExternalStyleSheet.Checked = True
		End If
	End Sub
	Private Sub buttonRunReport_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonRunReport.Click
		Dim report As SectionReport = Nothing
		'Select the report in the viewer.
		If radioButtonProductListReport.Checked Then
			report = New ProductsReport()
		ElseIf radioButtonCategoriesReport.Checked Then
			report = New CategoryReport()
		End If
		'Apply stylesheet on the report.
		Dim outputFolder As String
		outputFolder = New FileInfo(Me.GetType().Assembly.Location).DirectoryName & "\"
		Dim styleSheet As String = ""
		If radioButtonClassicStyle.Checked Then
			styleSheet = outputFolder & "Classic.reportstyle"
		ElseIf radioButtonColoredStyle.Checked Then
			styleSheet = outputFolder & "Colored.reportstyle"
		ElseIf externalStyleSheet <> "" Then
			styleSheet = externalStyleSheet
		End If
		If styleSheet <> "" Then
			report.LoadStyles(styleSheet)
		End If
		reportViewer.LoadDocument(report)
	End Sub
End Class

File: Section Reports/Layout/Stylesheets/C#/Stylesheets/StyleSheetsForm.cs
using System;
using System.IO;
using System.Windows.Forms;
namespace GrapeCity.ActiveReports.Samples.StyleSheets
{
	public partial class StyleSheetsForm : Form
	{
		private string _externalStyleSheet = string.Empty;
		public StyleSheetsForm()
		{
			InitializeComponent();
		}
		private void buttonRunReport_Click(object sender, EventArgs e)
		{
			SectionReport report = null;
			// Select the report in the Viewer.
			// 
			if (radioButtonProductListReport.Checked)
				report = new ProductsReport();
			else if (radioButtonCategoriesReport.Checked)
				report = new CategoryReport();
			// Apply stylesheet on the report.
			// 
			string outputFolder = new FileInfo(GetType().Assembly.Location).DirectoryName + "\\";
			string styleSheet = "";
			if (radioButtonClassicStyle.Checked)
				styleSheet = outputFolder + "Classic.reportstyle";
			else if (radioButtonColoredStyle.Checked)
				styleSheet = outputFolder + "Colored.reportstyle";
			else if (_externalStyleSheet != "")
				styleSheet = _externalStyleSheet;
			if (styleSheet != "")
			{
				report.LoadStyles(styleSheet);
			}
			reportViewer.LoadDocument(report);
		}
		private void buttonChooseExtStyle_Click(object sender, EventArgs e)
		{
			// Select the external stylesheet to apply on the report.
			// 
			FileDialog openFileDialog = new OpenFileDialog();
			
			openFileDialog.Filter = "Style sheet|*.reportstyle";
			
			
			openFileDialog.InitialDirectory = new FileInfo(GetType().Assembly.Location).DirectoryName;
			openFileDialog.CheckFileExists = true;
			if (openFileDialog.ShowDialog(this) == DialogResult.OK)
			{
				FileInfo styleSheetFile = new FileInfo(openFileDialog.FileName);
				_externalStyleSheet = styleSheetFile.FullName;
				
				radioButtonExternalStyleSheet.Text = "External stylesheet: " + styleSheetFile.Name;
				
				
				radioButtonExternalStyleSheet.Checked = true;
			}
		}
	}
}
